compaction/task_manager_module.cc

```diff
@@ -596,15 +596,25 @@ future<> table_reshaping_compaction_task_impl::run() {
 future<> shard_reshaping_compaction_task_impl::run() {
     auto& table = _db.local().find_column_family(_status.keyspace, _status.table);
     tasks::task_info info{_status.id, _status.shard};
+    std::unordered_map<size_t, std::unordered_set<sstables::shared_sstable>> sstables_grouped_by_compaction_group;
+    for (auto& sstable : _dir.get_unshared_local_sstables()) {
+        auto compaction_group_id = table.get_compaction_group_id_for_sstable(sstable);
+        sstables_grouped_by_compaction_group[compaction_group_id].insert(sstable);
+    }
 
```

Comment:
I think we can simplify it if the original function `shard_reshaping_compaction_task_impl::run()` becomes:

`shard_reshaping_compaction_task_impl::reshape_compaction_group(std::unordered_set<sstables::shared_sstable> ssts);`

And the original `shard_reshaping_compaction_task_impl::run()`  is simply mapping sstables in _dir.get_unshared_local_sstables() to groups, and calling reshape_compaction_group(...) on each.

Also, there's no need to reshape in parallel, as compaction manager will serialize each reshape anyway. So we'd only be queueing work that won't be dispatched anyway.